services/src/main/java/org/keycloak/services/resources/admin/UserResource.java

```diff
@@ -188,7 +188,11 @@ public Response updateUser(final UserRepresentation rep) {
             if (rep.isEnabled() != null && rep.isEnabled()) {
                 UserLoginFailureModel failureModel = session.loginFailures().getUserLoginFailure(realm, user.getId());
                 if (failureModel != null) {
-                    failureModel.clearFailures();
+                    session.loginFailures().removeUserLoginFailure(realm, user.getId());
```

Comment:
Previously, we were issuing an update to clear the login failure state. Changing now to remove it from the cache. I can't see why we have two distinct operations to unlock users when brute force is enabled so I'm changing here to actually remove the entry from the cache.

Reply:
@ahus1 Do you have any idea or anything against about changing to remove the entry from the cache rather than just update it?

Reply:
On the Infinispan level, with ISPN 15 a remove should be safe as it will propagate to the remote site even if there is no entry in the remote cache. This wasn't the case with ISPN 14.

On the functional level, the old code didn't clear numFailures and failedLoginNotBefore, not sure how much a difference that makes.

Reply:
I mean https://github.com/keycloak/keycloak/blob/main/model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/entities/LoginFailureEntity.java#L111. The values are set to `0`, that is why I don't see any difference when just removing the entry.